crates/webidl/src/lib.rs

```diff
@@ -924,3 +928,90 @@ impl<'src> WebidlParse<'src, (&'src str, &'src mut backend::ast::Module)> for we
         Ok(())
     }
 }
+
+// tons more data for what's going on here at
+// https://www.w3.org/TR/WebIDL-1/#idl-dictionaries
+impl<'src> WebidlParse<'src, ()> for weedle::DictionaryDefinition<'src> {
+    fn webidl_parse(
+        &'src self,
+        program: &mut backend::ast::Program,
+        first_pass: &FirstPassRecord<'src>,
+        (): (),
+    ) -> Result<()> {
+        if util::is_chrome_only(&self.attributes) {
+            return Ok(());
+        }
+
+        let mut fields = Vec::new();
+        if !push_members(first_pass, self.identifier.0, &mut fields) {
+            return Ok(())
+        }
+
+        program.dictionaries.push(ast::Dictionary {
+            vis: public(),
+            name: rust_ident(&camel_case_ident(self.identifier.0)),
+            fields,
+        });
+
+        return Ok(());
+
+        fn push_members<'src>(
+            data: &FirstPassRecord<'src>,
+            dict: &'src str,
+            dst: &mut Vec<ast::DictionaryField>,
+        ) -> bool {
+            let dict_data = &data.dictionaries[&dict];
+            let definition = dict_data.definition.unwrap();
+
+            // > The order of the dictionary members on a given dictionary is
+            // > such that inherited dictionary members are ordered before
+            // > non-inherited members ...
+            if let Some(parent) = &definition.inheritance {
+                if !push_members(data, parent.identifier.0, dst) {
+                    return false
+                }
+            }
+
+            // > ... and the dictionary members on the one dictionary
+            // > definition (including any partial dictionary definitions) are
+            // > ordered lexicographically by the Unicode codepoints that
+            // > comprise their identifiers.
+            let start = dst.len();
+            let members = definition.members.body.iter();
+            let partials = dict_data.partials.iter().flat_map(|d| &d.members.body);
+            for member in members.chain(partials) {
+                match mkfield(data, member) {
+                    Some(f) => dst.push(f),
+                    None => {
+                        warn!(
+                            "unsupported dictionary field {:?}",
+                            (dict, member.identifier.0),
+                        );
+                        // If this is required then we can't support the
+                        // dictionary at all, but if it's not required we can
+                        // avoid generating bindings for the field and keep
+                        // going otherwise.
+                        if member.required.is_some() {
+                            return false
+                        }
+                    }
+                }
+            }
+            dst[start..].sort_by_key(|f| f.name.clone());
```

Comment:
This is sorting by the snake case identifier, rather than the original identifier, which could produce the wrong result in edge cases:
```webidl
dictionary Example {
    boolean fooBaz;
    boolean foo_bar;
};
```
That is correctly ordered per the spec, but this line will reorder it to `[foo_bar, foo_baz]`.